lib/notesDisplay.ml

```diff
@@ -302,46 +306,43 @@ let print_what_links conf base fnotes =
   Option.iter (print_linked_list conf base) (List.assoc_opt fnotes db);
   Hutil.trailer conf
 
+let read_notes_from_conf conf base =
+  let fnotes =
+    match p_getenv conf.env "f" with
+    | Some f -> if NotesLinks.check_file_name f <> None then f else ""
+    | None -> ""
+  in
+  read_notes base fnotes
+
+let print_json conf base =
+  let nenv, s = read_notes_from_conf conf base in
+  let s =
+    match List.assoc "TYPE" nenv with
+    | "album" | "gallery" -> Notes.safe_gallery conf s
+    | (exception Not_found) | _ -> s
```

Comment:
This behavior is strange. The printer does not output a valid JSON message if the request does not contain a valid `TYPE` variable. I think that we should print an error in JSON format?